assets/js/base/components/loading-component/index.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Spinner } from 'wordpress-components';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+const LoadingComponent = ( {
+	children,
+	className,
+	screenReaderLabel,
+	showSpinner = false,
+} ) => {
+	return (
+		<div
+			className={ classNames( className, 'wc-block-loading-component' ) }
+		>
+			{ showSpinner && <Spinner /> }
+			<div
+				className="wc-blocks-loading-component__children"
+				aria-hidden={ true }
+			>
+				{ children }
+			</div>
+			<span className="screen-reader-text">
+				{ screenReaderLabel ||
+					__( 'Loading…', 'woo-gutenberg-products-block' ) }
```

Comment:
Is it good practice to use real entities in translated strings? Or should we use a HTML character code instead? `&hellip;`

Reply:
> Is it good practice to use real entities in translated strings? Or should we use a HTML character code instead? &hellip;

I think this is the UTF-8 character (Horizontal Ellipsis, U+2026) here which should be sufficient? Interestingly enough, Albert has an in progress pull at #1690 that addresses this (see also [here](https://github.com/Automattic/wp-calypso/blob/master/packages/eslint-plugin-wpcalypso/docs/rules/i18n-ellipsis.md)).

Reply:
Gutenberg is [using ellipsis in the same way](https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/post-publish-button/label.js#L23
) and, as @nerrad said, there is a Calypso ESLint rule to force using `…` in translations strings.

I didn't do a full test to see how that will render in translate.wordpress.org, but assuming Gutengberg and Calypso are using it, I think it's safe to use it ourselves too.